src/include/liburing.h

```diff
@@ -987,6 +987,15 @@ IOURINGINLINE void io_uring_prep_link(struct io_uring_sqe *sqe,
 	io_uring_prep_linkat(sqe, AT_FDCWD, oldpath, AT_FDCWD, newpath, flags);
 }
 
+IOURINGINLINE void io_uring_prep_msg_ring_cqe_flags(struct io_uring_sqe *sqe,
+					  int fd, unsigned int len, __u64 data,
+					  unsigned int flags, unsigned int cqe_flags)
+{
+	io_uring_prep_rw(IORING_OP_MSG_RING, sqe, fd, NULL, len, data);
+	sqe->rw_flags = IORING_MSG_RING_FLAGS_PASS | flags;
```

Comment:
Should probably use ->msg_ring_flags

Reply:
Thanks. Should we do the same for `io_uring_prep_msg_ring()`, which is using `        sqe->rw_flags` as well?

```
IOURINGINLINE void io_uring_prep_msg_ring(struct io_uring_sqe *sqe, int fd,
                                          unsigned int len, __u64 data,
                                          unsigned int flags)
{
        io_uring_prep_rw(IORING_OP_MSG_RING, sqe, fd, NULL, len, data);
        sqe->rw_flags = flags;
}

```

Reply:
https://github.com/axboe/liburing/pull/771 contains the fix.